test/llvm-intrinsics/bitreverse_small_type.ll

```diff
@@ -0,0 +1,143 @@
+;; Check that llvm.bitreverse.* intrinsics are lowered to emulation functions for
+;; 2/4-bit types.
+
+; The scalar calls can be generated from:
+;
+;   using uint2_t = unsigned _BitInt(2);
+;   using uint4_t = unsigned _BitInt(4);
+;   uint4_t reverse4(uint4_t a) {
+;   a = ((0x5 & a) << 1) | (0x5 & (a >> 1));
+;   return (a << 2) | (a >> 2);
+;   }
+;   uint2_t reverse2(uint2_t a) {
+;   return (a << 1) | (a >> 1);
+;   }
+;
+; with the command clang++ -O2 -emit-llvm -S bitreverse.c.
+; The vector cases cannot be generated by clang++ since elements smaller than a char
+; are not supported in vectors.  The vector cases are lowered for completeness.
+
+; RUN: llvm-as %s -o %t.bc
+; SPV_KHR_bit_instructions extension was not enabled so BitReverse must not be generated
```

Comment:
Think the run statement is self-explanatory. Not sure if we need this comment.

Reply:
Removed comment